hera_sim/simulate.py

```diff
@@ -754,64 +761,29 @@ def _get_model_parameters(model):
-                # function or a class instance
-                return component, False
-            if not isinstance(component, str):
-                # TODO: update this error message to reflect the
-                # change in allowed component types
-                raise TypeError(
-                    "``component`` must be either a class which "
-                    "derives from ``SimulationComponent`` or an "
-                    "instance of a callable class, or a function, "
-                    "whose signature is:\n"
-                    "func(lsts, freqs, *args, **kwargs)\n"
-                    "If it is none of the above, then it must be "
-                    "a string which corresponds to the name of a "
-                    "``hera_sim`` class or an alias thereof."
+    def _get_component(
+        component: [str, Type[SimulationComponent], SimulationComponent]
+    ) -> Tuple[Union[SimulationComponent, Type[SimulationComponent]], bool]:
+        """Given an input component, normalize the output to be either a class or instance.
+
+        """
+        if np.issubclass_(component, SimulationComponent):
+            return component, True
+        elif isinstance(component, str):
+            try:
+                return get_model(component), True
+            except KeyError:
+                raise ValueError(
+                    f"The model '{component}' does not exist. The following models are "
+                    f"available: \n{print_all_components()}."
                 )
-
-            # keep track of all known aliases in case desired
-            # component isn't found in the search
-            all_aliases = []
-            for registry in SimulationComponent.__subclasses__():
-                for model in registry.__subclasses__():
-                    aliases = (model.__name__,)
-                    aliases += getattr(model, "_alias", ())
-                    aliases = [alias.lower() for alias in aliases]
-                    for alias in aliases:
-                        all_aliases.append(alias)
-                    if component.lower() in aliases:
-                        return model, True
-
-            # if this part is executed, then the model wasn't found, so
-            string_of_aliases = ", ".join(set(all_aliases))
-            raise UnboundLocalError(
-                f"The component '{component}' wasn't found. The "
-                f"following aliases are known: \n{string_of_aliases}\n"
-                "Please ensure that the component you are trying "
-                "to add is a subclass of a registry."
+        elif isinstance(component, SimulationComponent):
+            return component, False
+        else:
+            raise ValueError(
+                "The input type for the component was not understood. "
+                "Must be a string, or a class/instance of type 'SimulationComponent'. "
+                f"Available component models are:\n{print_all_components()}"
```

Comment:
This is much more informative than before, thank you! The last part of the error message isn't what's expected--see previous comment.

Reply:
Fixed as above.